lib/mongos.js

```diff
@@ -411,7 +411,12 @@ class Mongos extends EventEmitter {
       }
 
       // No process, just resolve
-      if (!self.process) return resolve();
+      if (!self.process || self.state === 'stopped') {
```

Comment:
@mbroadst As I said, I am **not** able to reproduce the *"addr already in use"* message, but this is a theory that I was testing. I kept noticing the message occurs for the `Mongos` server so I was looking at the differences between `Mongos` and `Server`. This was one of them.

Reply:
Hmm, I wonder if there is potential in the future to provide a base class for all of our managers for common functionality like this?

Reply:
:+1 Good to keep in mind for the future!

Reply:
@mbroadst Do you mind testing with this change (it's easy enough to make on any branch without having to pull this PR) to see if you can reproduce the *addr already in use* message?

I'm unable to reproduce, but I wasn't able to produce it in the first place!